src/modifierOnHitPoison.ts

```diff
@@ -0,0 +1,45 @@
+import { registerEvents, registerModifiers } from "./cards";
+import { poisonCardId } from "./cards/poison";
+import { getOrInitModifier } from "./cards/util";
+import * as Unit from './entity/Unit';
+import Underworld from './Underworld';
+
+// Applies (quantity) poison to a unit when hitting it
+export const onHitPoisonId = 'On Hit Poison';
+export default function registerOnHitPoison() {
+  registerModifiers(onHitPoisonId, {
+    description: 'Applies (quantity) poison to a unit when hitting it',
+    add: (unit: Unit.IUnit, underworld: Underworld, prediction: boolean, quantity: number = 1) => {
+      getOrInitModifier(unit, onHitPoisonId, { isCurse: false, quantity, keepOnDeath: true }, () => {
+        Unit.addEvent(unit, onHitPoisonId);
+      });
+
+      if (!prediction) {
+        updateTooltip(unit);
+      }
+    }
+  });
+  registerEvents(onHitPoisonId, {
+    onDealDamage: (damageDealer: Unit.IUnit, amount: number, underworld: Underworld, prediction: boolean, damageReciever?: Unit.IUnit) => {
+      const modifier = damageDealer.modifiers[onHitPoisonId];
+      if (modifier) {
+        // Poison is only applied if damage is dealt, not when healing
+        if (damageReciever && amount > 0) {
+          // Apply poison to the damage reciever
+          Unit.addModifier(damageReciever, poisonCardId, underworld, prediction, modifier.quantity, { sourceUnitId: damageDealer.id });
```

Comment:
I love this rune.  I think we could start to add some interesting spells that affect "Poisoned units" like how we do with drown.

Reply:
Yesss, that would be awesome! For common mechanics like poison and shield, I think it would be cool to have runes and modifiers that give you new ways of applying the effect, new reasons to apply the effect, and new interactions/synergies. 

It would be cool to see plague builds that help you apply and spread poison, but also give your poisons fun secondary effects like "your poisons now slow enemies"

A self-sustained thorns+lifesteal build that makes you near invulnerable to most regular/low level enemies.
A singletarget rend build that stacks more and more damage as you focus on one enemy, and grants some kind of onKill bounty reward or heal or something.
Crazy things like that

Reply:
Love this. Can you add some of these to the runes list spreadsheet please?